synapse/rest/client/v1/room.py

```diff
@@ -267,6 +268,226 @@ def on_PUT(self, request, room_id, event_type, txn_id):
         )
 
 
+class RoomBulkSendEventRestServlet(TransactionRestServlet):
+    def __init__(self, hs):
+        super().__init__(hs)
+        self.store = hs.get_datastore()
+        self.state_store = hs.get_storage().state
+        self.event_creation_handler = hs.get_event_creation_handler()
+        self.room_member_handler = hs.get_room_member_handler()
+        self.auth = hs.get_auth()
+
+    async def _send_marker_event_for_insertion(
+        self, requester, room_id, insertion_event_id, auth_event_ids
+    ):
+        marker_event_dict = {
+            "type": EventTypes.MSC2716_MARKER,
+            "sender": requester.user.to_string(),
+            "content": {
+                EventContentFields.MSC2716_PREV_INSERTION: insertion_event_id,
+            },
+            "room_id": room_id,
+        }
+
+        (event, _,) = await self.event_creation_handler.create_and_send_nonmember_event(
+            requester,
+            marker_event_dict,
+            # TODO: Do we need to use `self.auth.compute_auth_events(...)` to filter the `auth_event_ids`?
+            auth_event_ids=auth_event_ids,
+        )
+
+    def register(self, http_server):
+        # /rooms/$roomid/bulksend
+        PATTERNS = "/rooms/(?P<room_id>[^/]*)/bulksend"
```

Comment:
I'm wondering if we want to rename this endpoint to only be about importing historical messages. The endpoint has expanded a bit include some specialized logic with the "insertion", "marker", and setting `inherit_depth=True`. 

These specialized options could be made to be conditional if we want to also allow this to be used generically. [MSC3192](https://github.com/matrix-org/matrix-doc/pull/3192) was just created to propose a bulk state endpoint which has some overlap with this MSC if we want to make this endpoint generic

Reply:
I think it makes sense for this to be a specialised API.

Reply:
If this isn't meant to be a generic endpoint for inserting state / messages I suspect we should specialize it, otherwise it will be confusing. (Especially if MSC3192 is approved.)

Probably being consistent on batch vs. bulk for both endpoints makes sense though. :)